File: day1/solution2.py
import sys

def get_frequency(path):
    frequency = 0
    check = set()
    while frequency not in check:
        check.add(frequency)
        with open(path) as fh:
            for line in fh:
                check.add(frequency)
                sign = line[0]
                value = int(line[1:])
                if sign == '+':
                    frequency += value
                elif sign == '-':
                    frequency -= value
                else:
                    print("ERROR: We encountered an input that doesn't have a +/- as the first char: {}".format(sign))
                    sys.exit(1)
                if frequency in check:
                    return frequency
    return frequency

def main():
    print("First repeated frequency: {}".format(get_frequency("input.txt")))

if __name__ == '__main__':
    main()

File: day1/solution2.rs
use std::fs;
use std::io;
use std::io::BufRead;
use std::process;
use std::collections::HashSet;

fn main() {
    let mut check = HashSet::new();
    let mut frequency: i32 = 0;

    while !check.contains(&frequency) {
        let input = fs::File::open("input.txt")
            .expect("Something went wrong reading the file");
        let reader = io::BufReader::new(input);

        for line in reader.lines() {
            check.insert(frequency);
            frequency += line.unwrap().parse::<i32>().unwrap();
            if check.contains(&frequency) {
                println!("Final frequency: {}", frequency);
                process::exit(0);
            }
        }
    }
}

